builder/docker/step_commit.go

```diff
@@ -14,11 +14,17 @@ type StepCommit struct {
 }
 
 func (s *StepCommit) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
-	driver := state.Get("driver").(Driver)
-	containerId := state.Get("container_id").(string)
-	config := state.Get("config").(*Config)
 	ui := state.Get("ui").(packer.Ui)
+	config, ok := state.Get("config").(*Config)
+	if !ok {
```

Comment:
Added the type assertion checks as a safety net to halt the packer run if there is a conversion issue. Let me know if I should build on this a little more or just 86 entirely.